middleware/block_v3_only_roles.rb

```diff
@@ -36,20 +36,21 @@ def globally_authenticated?
       end
 
       def space_supporter_and_only_space_supporter?(current_user)
-        user_roles =
-          VCAP::CloudController::SpaceManager.select(Sequel.as(VCAP::CloudController::RoleTypes::SPACE_MANAGER, :type)).limit(1).where(user_id: current_user.id).
-          union(VCAP::CloudController::SpaceDeveloper.select(Sequel.as(VCAP::CloudController::RoleTypes::SPACE_DEVELOPER, :type)).limit(1).where(user_id: current_user.id)).
-          union(VCAP::CloudController::SpaceAuditor.select(Sequel.as(VCAP::CloudController::RoleTypes::SPACE_AUDITOR, :type)).limit(1).where(user_id: current_user.id)).
-          union(VCAP::CloudController::SpaceSupporter.select(Sequel.as(VCAP::CloudController::RoleTypes::SPACE_SUPPORTER, :type)).limit(1).where(user_id: current_user.id)).
-          union(VCAP::CloudController::OrganizationManager.select(Sequel.as(VCAP::CloudController::RoleTypes::ORGANIZATION_MANAGER,
-                                                                            :type)).limit(1).where(user_id: current_user.id)).
-          union(VCAP::CloudController::OrganizationBillingManager.select(Sequel.as(VCAP::CloudController::RoleTypes::ORGANIZATION_BILLING_MANAGER,
-                                                                                   :type)).limit(1).where(user_id: current_user.id)).
-          union(VCAP::CloudController::OrganizationAuditor.select(Sequel.as(VCAP::CloudController::RoleTypes::ORGANIZATION_AUDITOR,
-                                                                            :type)).limit(1).where(user_id: current_user.id)).
-          all
-
-        user_roles.count == 1 && user_roles[0][:type] == 'space_supporter'
+        return false if VCAP::CloudController::User.
+                        where do
+                          Sequel.or(
+                            [
+                              [VCAP::CloudController::SpaceDeveloper.select(1).where(user_id: current_user.id).exists, true],
+                              [VCAP::CloudController::OrganizationManager.select(1).where(user_id: current_user.id).exists, true],
+                              [VCAP::CloudController::SpaceManager.select(1).where(user_id: current_user.id).exists, true],
+                              [VCAP::CloudController::SpaceAuditor.select(1).where(user_id: current_user.id).exists, true],
+                              [VCAP::CloudController::OrganizationAuditor.select(1).where(user_id: current_user.id).exists, true],
+                              [VCAP::CloudController::OrganizationBillingManager.select(1).where(user_id: current_user.id).exists, true]
+                            ]
+                          )
+                        end.any?
+
+        VCAP::CloudController::SpaceSupporter.select(1).where(user_id: current_user.id).any?
```

Comment:
Here the `.select(1)` is not needed. It's added by the `.any?`.

Reply:
Good point